File: src/change/imageUtil.py
from .wpcImage import WpcImage
from typing import List
from PIL import Image

def getImageWithHighestHeight(images: List[WpcImage]):
    highestHeightImage = None
    for image in images:
        if highestHeightImage is None:
            highestHeightImage = image
        if image.size.height > highestHeightImage.size.height:
            highestHeightImage = image
    return highestHeightImage

def getImageWithHighestWidth(images: List[WpcImage]):
    highestWidthImage = None
    for image in images:
        if highestWidthImage is None:
            highestWidthImage = image
        if image.size.width > highestWidthImage.size.width:
            highestWidthImage = image
    return highestWidthImage

def getImagesWidthSum(images: List[WpcImage]):
    width = 0
    for image in images:
        width += image.size.width
    return width

def checkIfImageAlreadyExist(newImage:Image, images:List[WpcImage]):
    for image in images:
        if image.getFullName() == newImage.getFullName():
            return True
    return False

File: Util/ImageUtil.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WallpaperChanger.Shared;

namespace WallpaperChanger.Util
{
    static class ImageUtil
    {
        public static WpcImage? getImageWithHighestHeight(List<WpcImage> wpcImages)
        {
            WpcImage? highestHeightImage = null;
            foreach (var wpcImage in wpcImages)
            {
                if (highestHeightImage == null || wpcImage.size.height > highestHeightImage.size.height)
                    highestHeightImage = wpcImage;
            }
            return highestHeightImage;
        }

        public static WpcImage? getImageWithHighestWidth(List<WpcImage> wpcImages)
        {
            WpcImage? highestWidthImage = null;
            foreach (var wpcImage in wpcImages)
            {
                if (highestWidthImage == null || wpcImage.size.width > highestWidthImage.size.width)
                    highestWidthImage = wpcImage;
            }
            return highestWidthImage;
        }

        public static int getImagesWidthSum(List<WpcImage> wpcImages)
        {
            int widthSum = 0;
            foreach (var wpcImage in wpcImages)
                widthSum += wpcImage.size.width;
            return widthSum;
        }

        public static bool checkIfImageAlreadyExist(WpcImage checkImage, List<WpcImage> wpcImages)
        {
            foreach (var wpcImage in wpcImages)
            {
                if (checkImage.getFullName().Equals(wpcImage.getFullName())) 
                    return true;
            }
            return false;
        }
    }
}
